Neuro/Vision/VisionHandler.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Il2CppInterop.Runtime.InteropTypes.Arrays;
+using Neuro.Recording.DataStructures;
+using Neuro.Vision.DataStructures;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+
+namespace Neuro.Vision;
+
+// TODO: Refactor this entire class
+[RegisterInIl2Cpp]
+public class VisionHandler : MonoBehaviour
+{
+    public VisionHandler(IntPtr ptr) : base(ptr) { }
+
+    public Vector2 DirectionToNearestBody { get; set; }
+    public readonly Dictionary<byte, PlayerRecord> PlayerRecords = new(); // TODO: Store this data using a monobehaviour on the player
+
+    // TODO: Handle players disconnecting
+    private readonly List<DeadBody> deadBodies = new();
+    private readonly Dictionary<byte, PlayerControl> playerControls = new(); // TODO: Use GameData.GetPlayerById
+    private readonly Dictionary<PlayerControl, LastSeenPlayer> playerLocations = new(); // TODO: Don't use PlayerControls as dictionary keys,
+    // TODO#2: These dictionaries arent cleaned after games
+
+    private float roundStartTime; // in seconds
+
+    public void FixedUpdate()
+    {
+        if (!ShipStatus.Instance) return;
+        if (MeetingHud.Instance) return;
+
+        UpdateDeadBodiesVision();
+        UpdateNearbyPlayersVision();
+    }
+
+    public void StartTrackingPlayer(PlayerControl player)
+    {
+        // TODO: I don't like this
+
+        Il2CppArrayBase<PlayerControl> @base = PlayerControl.AllPlayerControls.ToArray();
+        for (int i = 0; i < @base.Count; i++)
+        {
+            PlayerControl playerControl = @base[i];
+            if (!playerControls.ContainsKey(playerControl.PlayerId))
+                playerControls.Add(playerControl.PlayerId, playerControl);
+
+            PlayerRecords.Add(playerControl.PlayerId, new PlayerRecord(-1, new MyVector2(0, 0)));
```

Comment:
Fix NRE spam when first loading the lobby
```suggestion
            PlayerRecords.TryAdd(playerControl.PlayerId, new PlayerRecord(-1, new MyVector2(0, 0)));
```